src/neuroconv/datainterfaces/ecephys/openephys/openephysanaloginterface.py

```diff
@@ -0,0 +1,203 @@
+
+        session_start_time = _get_session_start_time(element=self._xml_root)
+        if session_start_time is not None:
+            metadata["NWBFile"].update(session_start_time=session_start_time)
+
+        return metadata
+
+    def get_channel_names(self) -> list[str]:
+        """
+        Get a list of analog channel names from the recording extractor.
+
+        Returns
+        -------
+        list of str
+            The names of all analog channels in the OpenEphys recording.
+        """
+        return list(self.analog_channel_ids)
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        iterator_type: Optional[str] = "v2",
+        iterator_opts: Optional[dict] = None,
+        always_write_timestamps: bool = False,
+    ):
+        """
+        Add analog channel data to an NWB file.
+
+        Parameters
+        ----------
+        nwbfile : NWBFile
+            The NWB file to which the analog data will be added
+        metadata : Optional[dict], default: None
+            Metadata dictionary with device information. If None, uses default metadata
+        stub_test : bool, default: False
+            If True, only writes a small amount of data for testing
+        iterator_type : Optional[str], default: "v2"
+            Type of iterator to use for data streaming
+        iterator_opts : Optional[dict], default: None
+            Additional options for the iterator
+        always_write_timestamps : bool, default: False
+            If True, always writes timestamps instead of using sampling rate
+        """
+        from ....tools.spikeinterface.spikeinterface import (
+            _recording_traces_to_hdmf_iterator,
+        )
+
+        if metadata is None:
+            metadata = self.get_metadata()
+
+        # Add devices
+        device_metadata = metadata.get("Devices", [])
+        for device in device_metadata:
+            if device["name"] not in nwbfile.devices:
+                nwbfile.create_device(**device)
+
+        if stub_test:
+            end_time = self.recording_extractor.get_end_time()
+            end_time = min(end_time, 0.100)
+            recording = self.recording_extractor.time_slice(start_time=0, end_time=end_time)
+        else:
+            recording = self.recording_extractor
+
+        channel_names = recording.get_property(key="channel_names")
+        segment_index = 0
+        analog_data_iterator = _recording_traces_to_hdmf_iterator(
+            recording=recording,
+            segment_index=segment_index,
+            iterator_type=iterator_type,
+            iterator_opts=iterator_opts,
+        )
+
+        name = self.source_data["time_series_key"]
+        description = f"Analog data from OpenEphys. Channels are {channel_names} in that order."
+        time_series_kwargs = dict(name=name, data=analog_data_iterator, unit="a.u.", description=description)
```

Comment:
Can we provide an option for the user to specify unit, conversion factor and offset in the metadata?

Reply:
Good point.

Question that I am facing as I am writing this type of interface. Do you think that sort of configuration is best in the metadata, in the `__init__` or, less likely, as conversion options? Bear in mind that they should be arrays as analog interfaces might have very different channels.

Independently of that, I am working on a way of propagating the units, gains and offests from Spikeinterface to here:

https://github.com/SpikeInterface/spikeinterface/pull/3739